src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayStateConditions = [];
+  let stateNew = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        stateNew = { ...stateNew, ...action.extraData };
+        break;
+      }
+
+      case 'removeProperties': {
+        stateNew = { ...stateNew };
+
+        for (const key of action.keysToRemove) {
+          delete stateNew[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        stateNew = {};
+        break;
+      }
+    }
```

Comment:
A `default` case should be added to the `switch` statement to handle any unexpected `action.type` values.